### PR TITLE
Moved swagger specs from resources to solutions folder.

### PR DESCRIPTION
**Pull request by vayada**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by vayada** [comment]
@ruowan  This PR is just moving swagger from resources folder to solutions folder as per Josephine's suggestion. There is not other change except moving specs from one location to another as per Josephine's suggestion. So, please help in merging this PR.

Source folders:
azure-rest-api-specs/tree/main/specification/resources/resource-manager/Microsoft.Solutions   (Main branch)
azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/solutions/resource-manager/Microsoft.Solutions (RPSaaSMaster)

Target Folder:
azure-rest-api-specs/specification/solutions/resource-manager/Microsoft.Solutions

**Comment by ruowan** [comment]
There are some validation errors, could you take a look and fix them?

**Comment by vayada** [comment]
> There are some validation errors, could you take a look and fix them?

@ruowan Fixed all validation errors and currently in touch with ARM API reviewer & breaking change reviewer. 

Can you look at sdk errors and let me know what needs to be done for that. Not sure what's causing duplicate key error as below.

"location":"specification/solutions/resource-manager/readme.md",
"tag":"[package-managedapplications-2021-07](https://github.com/Azure/azure-rest-api-specs/blob/442584aed7dbeac7befd0a85fb87e5d72102c80e/specification/solutions/resource-manager/readme.md#tag-package-managedapplications-2021-07)",
"details":"Syntax Error Encountered: Syntax error: duplicate key"

File: specification/solutions/resource-manager/readme.md [hunk 1]
```diff
@@ -28,10 +28,28 @@ These are the global settings for the managedApplication.
...
 tag: package-managedapplications-2021-07
+tag: package-managedapplications-2021-02
+tag: package-managedapplications-2020-08
+tag: package-managedapplications-2019-07
+tag: package-managedapplications-2018-09
+tag: package-managedapplications-2018-06
+tag: package-managedapplications-2018-03
+tag: package-managedapplications-2018-02
+tag: package-managedapplications-2017-12
+tag: package-managedapplications-2017-09
```
Comment (jianyexi) [inline specification/solutions/resource-manager/readme.md:39]: seems this is disallowed in yaml, you can defint multiple tag

**Comment by ruowan** [comment]
Hi, @jianyexi  Could you take a look at lintDiff, why there are so many lintDiff error?

**Comment by jianyexi** [comment]
@vayada the restructure causes the lintDiff failure, could you confirm all the versions except 2021-07-01 are coming from the old folder or branch and without any Api changes?

**Comment by vayada** [comment]
> @vayada the restructure causes the lintDiff failure, could you confirm all the versions except 2021-07-01 are coming from the old folder or branch and without any Api changes?
@jianyexi This contains the versions from old folder and some new api version specs which were missing and then I have corrected the validation failures in all the versions. So, its not just move now anymore. Please let me know what I can do to resolve these failures?

**Review by ruowan — APPROVED** [review]

File: specification/solutions/resource-manager/readme.md [hunk 2]
```diff
@@ -44,6 +44,87 @@ input-file:
...
 ```
 
+### Tag: package-managedapplications-2021-02
+
+These settings apply only when `--tag=package-managedapplications-2021-02` is specified on the command line.
+
+``` yaml $(tag) == 'package-managedapplications-2021-02'
+input-file:
+- Microsoft.Solutions/preview/2021-02-01-preview/managedapplications.json
+```
```
Comment (msyyc) [inline specification/solutions/resource-manager/readme.md:54]: tag name shall contain `preview` since it only contain files under preview folder

**Pull request merged**